src/include/hardware.h

```diff
@@ -45,6 +45,28 @@
 #define WT_READ_ONCE(v, val) WT_ORDERED_READ(v, val)
 #endif
 
+/*
+ * WT_WRITE_ONCE --
+ *
+ * Ensure a single write to memory in the source code produces a single write to memory in the
+ * compiled output.
+ *
+ * We can describe a scenario with a memory location 'a' which is read from into a local variable
```

Comment:
Alternate proposal:
"""
We can describe a scenario where a variable in memory is read to a local register, and then the register value is repeatedly updated before the final value is written back to memory. Without a WT_WRITE_ONCE wrapping the final write back to memory, the compiler is allowed to convert these register writes into writes to memory (see "register spilling") which can leak interim, invalid, state to other threads. We call these 'invented stores' in WiredTiger.
"""